src/relay/op/tensor/transform.h

```diff
@@ -65,31 +65,39 @@ bool ConcatenateRel(const Array<Type>& types,
   const int ndim = static_cast<int>(first->shape.size());
   const DataType dtype = first->dtype;
 
+  // Sanity check: axis
+  int axis = param->axis;
+  if (!(-ndim <= axis && axis < ndim)) {
+    throw relay::Error(RELAY_ERROR(
+      "concatenate only accepts `axis` in [-ndim, ndim)" <<
+      ", but got axis = " << axis <<
+      ", and ndim = " << ndim));
+  }
+  axis = axis < 0 ? ndim + axis : axis;
+
   for (const Type& ele : tensor_tuple->fields) {
     if (ele.as<IncompleteTypeNode>()) {
       return false;
     }
 
-    const auto& e = Downcast<TensorType>(ele);
+     const auto& e = Downcast<TensorType>(ele);
```

Comment:
wrong indent

Reply:
updated. Question, is there any convenient way to set code format in IDE like clion?

Reply:
clion should be able to add some plugins to format your code, you can configure it to vim mode and add clang-format as well.